yt_dlp/extractor/vimp.py

```diff
@@ -0,0 +1,152 @@
+                'display_id': 'Tutorial-zur-Nutzung-von-Adobe-Connect-aus-Veranstalter-Sicht',
+                'ext': 'mp4',
+            },
+        },
+        {
+            'url': 'https://www2.univ-sba.dz/video/Presentation-de-la-Faculte-de-droit-et-des-sciences-politiques-Journee-portes-ouvertes-202122/0183356e41af7bfb83d7667b20d9b6a3',
+            'info_dict': {
+                'url': 'https://www2.univ-sba.dz/getMedium/0183356e41af7bfb83d7667b20d9b6a3.mp4',
+                'id': '0183356e41af7bfb83d7667b20d9b6a3',
+                'title': 'Présentation de la Faculté de droit et des sciences politiques - Journée portes ouvertes 2021/22',
+                'description': 'Pr&eacute;sentation de la Facult&eacute; de droit et des sciences politiques - Journ&eacute;e portes ouvertes 2021/22',
+                'display_id': 'Presentation-de-la-Faculte-de-droit-et-des-sciences-politiques-Journee-portes-ouvertes-202122',
+                'ext': 'mp4',
+            }
+        },
+        {
+            'url': 'https://videocampus.sachsen.de/media/embed?key=fc99c527e4205b121cb7c74433469262',
+            'info_dict': {
+                'id': 'fc99c527e4205b121cb7c74433469262',
+                'title': 'Was ist selbstgesteuertes Lernen?',
+                'ext': 'mp4',
+            },
+        }
+    ]
+
+    def _real_extract(self, url):
+        host, video_id, tmp_id, display_id, embed_id = self._match_valid_url(url).group('host', 'id', 'tmp_id', 'display_id', 'embed_id')
+        webpage = self._download_webpage(url, video_id or tmp_id, fatal=False) or ''
+
+        if not video_id:
+            video_id = embed_id or self._html_search_regex(
+                rf'src="https?://{host}/media/embed\?key=([0-9a-f]+)&',
+                webpage, 'video_id')
+
+        # Title, description from page meta wouldn't be correct if the video is embedded
+        # so separating normal and embedded extraction
+        if not (display_id or tmp_id):
+            title = self._html_search_regex(r'<img[^>]*title="([^"<]+)"', webpage, 'title', fatal=False)
+            description = None
+        else:
+            title = self._html_search_meta(('og:title', 'twitter:title', 'title'), webpage, default=None)
+            description = self._html_search_meta(('og:description', 'twitter:description', 'description'), webpage, default=None)
+
+        # Try extraction of HLS first since it usually has multiple formats plus possibly subtitles
+        formats = subtitles = None
+        try:
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                f'https://{host}/media/hlsMedium/key/{video_id}/format/auto/ext/mp4/learning/0/path/m3u8',
+                video_id, 'mp4', 'm3u8_native', m3u8_id='hls')
+            self._sort_formats(formats)
+        except ExtractorError:
+            pass
```

Comment:
why not:
```py
formats, subtitles = self._extract...(..., fatal=False)
formats.append({'url': f'https://{host}/getMedium/{video_id}.mp4', ...})
self._sort_formats(formats)
```

Reply:
LGTM, but: I've set `fatal=False` but it yields `WARNING: [Vimp] Failed to download m3u8 information: HTTP Error 404: Not Found` which results in failure in tests.

Reply:
add an `expected_warnings` to the test definition.

Or, if you don't want the user to see this warning, catching the error makes sense. But (1) raise warning for errors other than `404`, (2) sort_formats should be outside the try block